sympy/assumptions/handlers/calculus.py

```diff
@@ -213,6 +213,19 @@ def _(expr, assumptions):
 def _(expr, assumptions):
     return True
 
-@FinitePredicate.register_many(Infinity, NegativeInfinity, NaN)
+@FinitePredicate.register_many(ComplexInfinity, Infinity, NegativeInfinity, NaN)
+def _(expr, assumptions):
+    return False
```

Comment:
In the old assumptions `nan.is_finite` gives `None` which I think is better.

Reply:
Done!